src/gordon_gcp/clients/auth.py

```diff
@@ -224,12 +237,14 @@ async def _execute_refresh_token_request(
                 logging.error(msg, exc_info=e)
                 raise exceptions.GCPHTTPResponseError(msg, resp.status)
 
-            response = await resp.json()
-            try:
-                self.token = response['access_token']
-            except KeyError:
-                msg = '[{request_id}] No access token in response.'
-                logging.error(msg)
-                raise exceptions.GCPAuthError(msg)
+            return await resp.json()
+
+    def _handle_refresh_token_response(self, response):
+        try:
+            self.token = response['access_token']
+        except KeyError:
```

Comment:
I realize this wasn't changed in this PR, but would you mind switching to `if 'access_token' in response`? Key lookups are less expensive than exceptions.